xsensmt/XsensMT.cpp

```diff
@@ -237,6 +266,13 @@ bool XsensMT::close()
     {
         m_sensorThread.join();
     }
+
+    // yDebug("xsensmt: Closing Xsens port %s.", m_portInfo.portName().toStdString().c_str());
+    // m_xsensControl->closePort(m_portInfo.portName().toStdString());
+    //
+    // yDebug("xsensmt: Freeing Xsens control object.");
+    // m_xsensControl->destruct();
```

Comment:
These are cleanup lines after closing the device, but I noticed they are being executed at a very early stage. And I couldn't understand why!

Reply:
Handled in https://github.com/robotology/yarp-device-xsensmt/pull/38/commits/dbd12e7bb354cecee2db0a29948c6accecd19348 with @traversaro 

Basically `XsensMT::close()` is called twice, once because it's called in the deconstructor of `XsensMT` and the second after sending a close signal to the YARP device.

The deconstructor is called as soon as `XsensMT` is constructed, probably even before `open()` is called.

We initialized `XsControl* m_xsensControl{nullptr}` to a `nullptr`, and called `m_xsensControl.deconstruct()` only if the `m_xsensControl` doesn't return `nullptr`. After deconstructing it, the pointer value is set to `nullptr` again.